src/main/java/ee/tuleva/onboarding/auth/principal/PrincipalService.java

```diff
@@ -17,7 +18,7 @@
 
   private final UserService userService;
 
-  public AuthenticatedPerson getFrom(Person person) {
+  public AuthenticatedPerson getFrom(Person person, Supplier<Optional<String>> phoneSupplier) {
```

Comment:
Why a supplier instead of just an `Optional<String> phoneNumber`?

Reply:
Good catch, feel free to simplify it :)